apps/search-poc/src/web-components/web-component.ts

```diff
@@ -0,0 +1,83 @@
+import React, { FunctionComponent } from 'react';
+import ReactDOM from 'react-dom';
+
+import { createNexusClient, NexusClient } from '@bbp/nexus-sdk';
+import { NexusProvider } from '@bbp/react-nexus';
+
+import { createShadowDomRootTree } from './shadow-dom-tools';
+import { setToken } from '../utils/auth';
+
+/**
+ * Generic web component class
+ *
+ * Derived class shoud define:
+ * * ReactComponent
+ * * nexusDeployment
+ * * reactComponentProps
+ *
+ * and then call .init to instantiate component
+ */
+export default class WebComponent extends HTMLElement {
+  ReactComponent: FunctionComponent<any> | string = '';
+  nexusDeployment: string | null = null;
+  reactComponentProps: { [prop: string]: any } = {};
+
+  private nexusClient: NexusClient | null = null;
+  private mountPoint: HTMLElement;
+  private initialized = false;
+
+  constructor() {
+    super();
+
+    const containerEl = createShadowDomRootTree();
+    this.mountPoint = containerEl.querySelector('div') as HTMLElement;
+
+    this.attachShadow({ mode: 'open' }).appendChild(containerEl);
+
+    const styleElClone = (window[
+      'getResourceDetailsStyleEl'
+    ]() as Node).cloneNode(true);
+    containerEl.appendChild(styleElClone);
+  }
+
+  init() {
```

Comment:
Write a note about why this init step is necessary

Reply:
Done, please let me know if it's not clear